### PR TITLE
docs: add date formats section for date picker

### PR DESCRIPTION
**Pull request by sissbruecker**
## Description

Adds section to date picker docs on how to customize the date format used for displaying and parsing dates. The section is split up into three sub-sections, each for a different mechanism either specific to the Flow or the web-component.

The custom date format feature is prerelease only. For testing the following dependency needs to be added to the pom.xml in order to use the snapshot that contains the new feature:
```
    <dependencies>
        ...

        <dependency>
            <groupId>com.vaadin</groupId>
            <artifactId>vaadin-date-picker-flow</artifactId>
            <version>22.0-SNAPSHOT</version>
        </dependency>
    </dependencies>
```

Related to https://github.com/vaadin/platform/issues/2277

Marked as draft until a v22 release containing the feature is available and used in the docs repo.

### CODE REVIEW

File: src/main/java/com/vaadin/demo/component/datepicker/DatePickerCustomFormat.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+import java.time.LocalDate;
+import java.time.ZoneId;
+
+@Route("date-picker-custom-format")
+public class DatePickerCustomFormat extends VerticalLayout {
+    public DatePickerCustomFormat() {
+        // tag::snippet[]
+        // Setup date picker with a single custom format `yyyy-MM-dd` for
+        // displaying dates and parsing user input
+        // Custom date formats are specified used the date pickers internationalization API
```
Comment (fhdhsni) [inline src/main/java/com/vaadin/demo/component/datepicker/DatePickerCustomFormat.java:19]:
```
using the date pickers internationalization API
```
or
```
via the date pickers internationalization API
```

File: articles/ds/components/date-picker/index.asciidoc [hunk 2]
```diff
@@ -35,6 +35,121 @@ The overlay opens when the field is clicked and/or any input is entered when the
...
 
 
+== Date Format
+
+The date picker can be configured to display dates and parse user input in a specific format.
+
+=== Using Java Locales (Java Only)
+
+By default, the date picker displays and parses dates using the user's locale (<<../../../flow/advanced/i18n-localization.asciidoc#locale-selection-for-new-session,reference>>).
+Alternatively setting a specific locale ensures that all users consistently see the same format.
```
Comment (fhdhsni) [inline articles/ds/components/date-picker/index.asciidoc:45]:
```suggestion
Alternatively, setting a specific locale ensures that all users consistently see the same format.
```

File: src/main/java/com/vaadin/demo/component/datepicker/DatePickerCustomFormat.java (hunk 1 above)
Comment (sissbruecker) [reply src/main/java/com/vaadin/demo/component/datepicker/DatePickerCustomFormat.java]: Done

File: articles/ds/components/date-picker/index.asciidoc (hunk 2 above)
Comment (sissbruecker) [reply articles/ds/components/date-picker/index.asciidoc]: Done

**Review by fhdhsni — APPROVED** [review]

**Pull request merged**